src/autosuggest/__tests__/autosuggest-and-token-group.test.tsx

```diff
@@ -0,0 +1,74 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+import React, { useState } from 'react';
+import { render } from '@testing-library/react';
+import createWrapper from '../../../lib/components/test-utils/dom';
+import Autosuggest from '../../../lib/components/autosuggest';
+import TokenGroup from '../../../lib/components/token-group';
+
+const wrapper = createWrapper();
+
+function getSelectedTokens() {
+  return wrapper
+    .findTokenGroup()!
+    .findTokens()
+    .map(token => token.findLabel().getElement().textContent!.trim());
+}
+
+function Demo() {
```

Comment:
Added a verbose test case to show a practical example where the `onSelect` handler can be used for good.